client/my-sites/controller.js

```diff
@@ -183,6 +185,18 @@ module.exports = {
 				// if sites have loaded, but siteID is invalid, redirect to allSitesPath
 				if ( sites.select( siteID ) ) {
 					onSelectedSiteAvailable();
+				} else if ( ( currentUser.visible_site_count !== sites.getVisible().length ) ) {
+					ReactDom.render( (
+						<Main>
+							<JetpackManageErrorPage template="waitForJetpackSite" />
+						</Main>
+					), document.getElementById( 'primary' ) );
+
+					setTimeout( () => {
```

Comment:
I started down that path almost exactly at first, then scrapped it for some reason 😔  

Oh -- I think immediately re-pinging for sites was sending more API requests before success than the artificial 1 second delay (it was about 5 to 1). If that's OK, I'll bring back this approach.